### ISSUE TITLE
Issue with scrolling up within a scrollable region

### ISSUE THREAD

**Issue by liamg** [liamg/aminal#19]
Scrolling up within a scrollable region in certain applications simply does not work. The line the cursor is on does update and scrolls content, but all other lines remain stationary and not updated.

**To Reproduce**
1. Run `htop`
2. Hold the down arrow and scroll to the end of the list of processes (you may need to shrink the window to make the list become scrollable)
3. Hold the up arrow and scroll back up. When the cursor reaches the top line of the scrollable region (list of processes), the list should begin to scroll, but it does not.

**Expected behavior**
In step 3 above, the list should scroll back toward the top.

**Additional context**
I think the origin mode being implemented may fix this issue? See https://vt100.net/docs/vt510-rm/DECOM.html

**Comment by liamg** [comment]
Dump of CSI codes received from htop after single scroll up with up arrow:
```
2018-10-22T12:46:59.239+0100	DEBUG	terminal/csi.go:292	CSI 0x72 (ESC[9;64r)
2018-10-22T12:46:59.239+0100	DEBUG	terminal/csi.go:292	CSI 0x48 (ESC[9;1H)
2018-10-22T12:46:59.239+0100	DEBUG	terminal/csi.go:292	CSI 0x72 (ESC[1;65r)
2018-10-22T12:46:59.239+0100	DEBUG	terminal/csi.go:292	CSI 0x48 (ESC[9;1H)
2018-10-22T12:46:59.239+0100	DEBUG	terminal/csi.go:292	CSI 0x6D (ESC[30m)
2018-10-22T12:46:59.239+0100	DEBUG	terminal/csi.go:292	CSI 0x6D (ESC[46m)
2018-10-22T12:46:59.239+0100	DEBUG	terminal/csi.go:292	CSI 0x4B (ESC[K)
2018-10-22T12:46:59.239+0100	DEBUG	terminal/csi.go:292	CSI 0x64 (ESC[10d)
2018-10-22T12:46:59.239+0100	DEBUG	terminal/csi.go:292	CSI 0x6D (ESC[39;49m)
2018-10-22T12:46:59.239+0100	DEBUG	terminal/csi.go:292	CSI 0x6D (ESC[m)
2018-10-22T12:46:59.239+0100	DEBUG	terminal/csi.go:292	CSI 0x6D (ESC[0;1m)
2018-10-22T12:46:59.239+0100	DEBUG	terminal/csi.go:292	CSI 0x6D (ESC[90m)
2018-10-22T12:46:59.239+0100	DEBUG	terminal/csi.go:292	CSI 0x6D (ESC[39m)
2018-10-22T12:46:59.239+0100	DEBUG	terminal/csi.go:292	CSI 0x6D (ESC[m)
2018-10-22T12:46:59.239+0100	DEBUG	terminal/csi.go:292	CSI 0x6D (ESC[31m)
2018-10-22T12:46:59.239+0100	DEBUG	terminal/csi.go:292	CSI 0x6D (ESC[39m)
2018-10-22T12:46:59.239+0100	DEBUG	terminal/csi.go:292	CSI 0x6D (ESC[m)
2018-10-22T12:46:59.239+0100	DEBUG	terminal/csi.go:292	CSI 0x6D (ESC[32m)
2018-10-22T12:46:59.239+0100	DEBUG	terminal/csi.go:292	CSI 0x6D (ESC[39m)
2018-10-22T12:46:59.239+0100	DEBUG	terminal/csi.go:292	CSI 0x6D (ESC[m)
2018-10-22T12:46:59.240+0100	DEBUG	terminal/csi.go:292	CSI 0x4B (ESC[K)
2018-10-22T12:46:59.240+0100	DEBUG	terminal/csi.go:292	CSI 0x48 (ESC[H)
```

**Comment by liamg** [comment]
Fixed in #20
